cmd/src/search_stream.go

```diff
@@ -321,25 +317,44 @@ const streamingTemplate = `
 `
 
 var streamSearchTemplateFuncs = map[string]interface{}{
-	"streamSearchHighlightMatch": func(query string, match streaming.EventLineMatch) string {
-		var highlights []highlight
-		if strings.Contains(query, "patterntype:structural") {
-			highlights = streamConvertMatchToHighlights(match, false)
-			return applyHighlightsForFile(match.Line, highlights)
+	"streamSearchHighlightMatch": func(match streaming.ChunkMatch) string {
+		var result []rune
+
+		addLineColumn := func(n int) []rune {
+			result = append(result, []rune("  ")...)
+			result = append(result, []rune(ansiColors["search-line-numbers"])...)
+			result = append(result, []rune(fmt.Sprintf("% 6d", match.ContentStart.Line+1+n))...)
+			result = append(result, []rune(ansiColors["nc"])...)
+			result = append(result, []rune(ansiColors["search-border"])...)
+			result = append(result, []rune(" |  ")...)
+			result = append(result, []rune(ansiColors["nc"])...)
+			return result
```

Comment:
this logic preserves the line column formatting, which is based off of seeing `\n`, and this logic is no longer easy to express in the template. on the plus side, the template becomes simpler